src/components/header/index.js

```diff
@@ -178,6 +178,18 @@ class Header extends PureComponent {
                         {...this.props}
                         {...this.state}
                         menuItems={navMain}
+                        toggleShowSubmenu={() => {
```

Comment:
if `!this.state.clickOutside` is true, it will never validate the `else if` which has the same `!this.state.clickOutside`, so you don't need to repeat that part.  You can shorten:

```suggestion
                        toggleShowSubmenu={() => {
                          this.setState({ showSubmenu: false });
  
                          if (!this.state.clickOutside || !this.props.fullScreen) {
                             this.setState({ showSubmenu: true });
                          } 
                        }}
```

Reply:
nice catch!, updated 👍